docs/usage/instance_deployment.md

```diff
@@ -0,0 +1,210 @@
+      {"apiVersion":"node.sapcloud.io/v1alpha1","kind":"InstanceDeployment","metadata":{"annotations":{},"name":"test-instance-deployment","namespace":""},"spec":{"minReadySeconds":200,"replicas":3,"selector":{"matchLabels":{"test-label":"test-label"}},"strategy":{"rollingUpdate":{"maxSurge":1,"maxUnavailable":1},"type":"RollingUpdate"},"template":{"metadata":{"labels":{"test-label":"test-label"}},"spec":{"class":{"kind":"AWSInstanceClass","name":"test-aws"}}}}}
+  clusterName: ""
+  creationTimestamp: 2017-12-27T08:55:56Z
+  generation: 0
+  initializers: null
+  name: test-instance-deployment
+  namespace: ""
+  resourceVersion: "12634168"
+  selfLink: /apis/node.sapcloud.io/v1alpha1/test-instance-deployment
+  uid: c0b488f7-eae3-11e7-a6c0-828f843e4186
+spec:
+  minReadySeconds: 200
+  replicas: 3
+  selector:
+    matchLabels:
+      test-label: test-label
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+    type: RollingUpdate
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        test-label: test-label
+    spec:
+      class:
+        kind: AWSInstanceClass
+        name: test-aws
+status:
+  availableReplicas: 3
+  conditions:
+  - lastTransitionTime: 2017-12-27T08:57:22Z
+    lastUpdateTime: 2017-12-27T08:57:22Z
+    message: Deployment has minimum availability.
+    reason: MinimumReplicasAvailable
+    status: "True"
+    type: Available
+  readyReplicas: 3
+  replicas: 3
+  updatedReplicas: 3
+```
+
+## Health monitoring
+
+Health monitor is also applied similar to how its described for [instance-sets](instance_set.md)
+
+## Update your instances 
+
+Let us consider the scenario where you wish to update all nodes of your cluster from t2.xlarge instances to m4.xlarge instances. Assume that your current *test-aws* has its **spec.instanceType: t2.xlarge** and your deployment *test-instance-deployment* points to this AWSInstanceClass. 
+
+#### Inspect existing cluster configuration
+
+- Check Nodes present in the cluster 
+```bash
+$  kubectl get nodes
+NAME                                          STATUS    AGE       VERSION
+ip-10-250-20-19.eu-west-1.compute.internal    Ready     1m        v1.8.0
+ip-10-250-27-123.eu-west-1.compute.internal   Ready     1m        v1.8.0
+ip-10-250-31-80.eu-west-1.compute.internal    Ready     1m        v1.8.0
+``` 
+- Check Node Controller Manager instance-sets in the cluster. You will notice one instance-set backing your instance-deployment
+```bash
+$ NAME                                  KIND
+test-instance-deployment-5bc6dd7c8f   InstanceSet.v1alpha1.node.sapcloud.io
+```
+- Login to your cloud provider (AWS) and in your VM management console, you will find N VMs created of type t2.xlarge.
+
+#### Perform a rolling update
+
+To update this instance-deployment VMs to m4.xlarge, we would do the following,
+
+- Copy your existing aws-instance-class.yaml 
+```bash
+$ cp kubernetes/aws-instance-class.yaml kubernetes/aws-instance-class-new.yaml
```

Comment:
why not modify it in place using the same file name ? If you do this for real you would put that in a git repository to track changes, right ?

Reply:
This exercise is used for demonstration of updating nodes from one version to another. So we maintain two files - one the original and the modified, at a later point the older file maybe required to have a look at the original version of the nodes. That was the idea behind having a copy. I hope its alright to have two versions of the same file for this reason ?

Reply:
ok